core/src/main/java/org/apache/accumulo/fate/ZooStore.java

```diff
@@ -478,6 +480,18 @@ public Serializable getProperty(long tid, String prop) {
     }
   }
 
+  @Override
+  public String getTimestamp(long tid) {
+    verifyReserved(tid);
+
+    try {
+      Stat stat = zk.getZooKeeper().exists(getTXPath(tid), false);
+      return new Date(stat.getCtime()).toString();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
```

Comment:
Consider returning an empty string (or N/A, or unavailable?) instead of throwing an exception?  This is for information and continuing without this data point (especially if the exception was transient) may allow other operations to continue if they can.  What is returned versus what is displayed may want to consider a user wanting to parse the output (say with awk or sort) along what is clear to a user.

Reply:
I can see the value in this. I was trying to match the other functions in this class with the exception handling but I can look into this more.

Reply:
Specifically - if NoNodeExists could possibly be thrown (say it completed asynchronously while doing the gathering) then a blank instead of throwing an exception would allow other values to be displayed.

Reply:
I made this change in the latest commit. I went with "ERROR" but I am open to something else or a more specific message depending on what actual exception occurred.